src/common/idle_inhibitor.cpp

```diff
@@ -0,0 +1,231 @@
+#endif
+}
+
+#if defined(_WIN32)
+void IdleInhibitor::InhibitIdle() {
+    inhibiting_idle = true;
+    SetThreadExecutionState(ES_CONTINUOUS | ES_SYSTEM_REQUIRED | ES_DISPLAY_REQUIRED);
+}
+
+void AllowIdle() {
+    inhibiting_idle = false;
+    SetThreadExecutionState(ES_CONTINUOUS);
+}
+
+#elif defined(__unix__)
+void IdleInhibitor::ResetScreensaver() {
+    auto poll_time = std::chrono::steady_clock::now() + std::chrono::seconds(10);
+    for (;;) {
+        std::unique_lock<std::mutex> lk(cond_mutex);
+        poll_cond.wait_until(lk, poll_time, [this]() { return !inhibiting_idle; });
+        if (!inhibiting_idle)
+            return;
+
+        if (fork() == 0)
+            execlp("xdg-screensaver", "xdg-screensaver", "reset", nullptr);
+
+        poll_time += std::chrono::seconds(10);
+    }
+}
+
+void IdleInhibitor::ConnectDBus() {
+    if (g_dbus.initialized) {
+        dbus_connection = g_dbus.bus_get(DBUS_BUS_SESSION, nullptr);
+
+        if (!dbus_connection) {
+            LOG_WARNING(Common, "Failed to connect to session bus");
+        }
+    }
+}
+
+void IdleInhibitor::InhibitIdle() {
+    if (inhibiting_idle)
+        return;
+
+    inhibiting_idle = true;
+
+    if (dbus_connection) {
+        DBusMessage* inhibit_call = g_dbus.message_new_method_call(
+            "org.freedesktop.ScreenSaver", "/org/freedesktop/ScreenSaver",
+            "org.freedesktop.ScreenSaver", "Inhibit");
+        if (!inhibit_call) {
+            LOG_ERROR(Common, "Failed to create Inhibit call");
+            return;
+        }
+
+        if (!g_dbus.message_append_args(inhibit_call, DBUS_TYPE_STRING, &g_app, DBUS_TYPE_STRING,
+                                        &g_reason, DBUS_TYPE_INVALID)) {
+            g_dbus.message_unref(inhibit_call);
+            LOG_ERROR(Common, "Failed to append args to Inhibit call");
+            return;
+        }
+
+        DBusMessage* reply = g_dbus.connection_send_with_reply_and_block(
+            dbus_connection, inhibit_call, DBUS_TIMEOUT_USE_DEFAULT, nullptr);
+        g_dbus.message_unref(inhibit_call);
+
+        if (reply) {
+            if (!g_dbus.message_get_args(reply, nullptr, DBUS_TYPE_UINT32, &inhibit_cookie)) {
+                LOG_ERROR(Common, "Failed to get cookie from Inhibit reply");
+                inhibit_cookie = 0;
+            }
+        } else {
+            LOG_WARNING(Common, "Inhibit call failed");
+        }
+    }
+
+    if (inhibit_cookie == 0) {
+        LOG_WARNING(Common, "Falling back to xdg-screensaver polling thread");
+        poll_thread = std::make_unique<std::thread>(&Common::IdleInhibitor::ResetScreensaver, this);
+    }
+}
+
+void IdleInhibitor::AllowIdle() {
+    if (!inhibiting_idle)
+        return;
+
+    cond_mutex.lock();
+    inhibiting_idle = false;
+    cond_mutex.unlock();
+
+    if (poll_thread && poll_thread->joinable()) {
+        poll_cond.notify_all();
+        poll_thread->join();
+        poll_thread = nullptr;
```

Comment:
```suggestion
        poll_thread.reset();
```

Makes the deallocation here a little more explicit.